kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/EvalHelper.java

```diff
@@ -38,8 +38,25 @@
 public class EvalHelper {
     public static final Logger LOG = LoggerFactory.getLogger( EvalHelper.class );
 
+    private static String removeBlankSpaces(String variableName) {
+        StringBuilder sb = new StringBuilder(variableName);
+        int j = 0;
+        boolean lastSpace = false;
+        for (int i = 0; i < sb.length(); i++) {
+            if (!Character.isWhitespace(sb.charAt(i))) {
+                sb.setCharAt(j++, sb.charAt(i));
+                lastSpace = false;
+            } else if (!lastSpace) {
+                sb.setCharAt(j++, sb.charAt(i));
+                lastSpace = true;
+            }
+        }
+        sb.delete(j, sb.length());
+        return sb.toString();
+    }
+
     public static String normalizeVariableName(String name) {
-        return name.replaceAll( "\\s+", " " );
```

Comment:
Thank you both for your comments. I did a quick test here profiling the 3 alternatives. For 1 million calls, here are the results on my machine:

1. using the current string.replaceAll() call: 0.305s
2. using the precompiled pattern suggested by @jhrcek : 0.106s
3. using the static method suggested by @marekbubala : 0.145s

I will go ahead and replace the current call by the precompiled pattern suggested by @jhrcek.

Thank you!